internal/mode/static/state/graph/nginxproxy.go

```diff
@@ -0,0 +1,151 @@
+	// Source is the corresponding NginxProxy proxy resource.
+	Source *ngfAPI.NginxProxy
+	// Tracing holds the tracing configuration.
+	Tracing *Tracing
+	// Conditions holds the conditions for the NginxProxy.
+	Conditions []conditions.Condition
+	// Valid indicates whether the NginxProxy Spec is valid.
+	Valid bool
+}
+
+// Tracing represents Otel tracing configuration for the dataplane.
+type Tracing struct {
+	// Endpoint specifies the address of OTLP/gRPC endpoint that will accept telemetry data.
+	Endpoint string
+	// ServiceName is the “service.name” attribute of the OTel resource.
+	ServiceName string
+	// Interval specifies the tracing interval.
+	Interval string
+	// BatchSize specifies the maximum number of spans to be sent in one batch per worker.
+	BatchSize int32
+	// BatchCount specifies the number of pending batches per worker, spans exceeding the limit are dropped.
+	BatchCount int32
+	// Enabled specifies if tracing is enabled.
+	Enabled bool
+}
+
+func buildNginxProxyConfig(
+	nps map[types.NamespacedName]*ngfAPI.NginxProxy,
+	gc *v1beta1.GatewayClass,
+	gw *v1beta1.Gateway,
+) *NginxProxy {
+	gwNsName := "unknown"
+	if gw != nil {
+		gwNsName = gw.Namespace + "/" + gw.Name
+	}
+	if gc != nil {
+		ref := gc.Spec.ParametersRef
+		if ref != nil && ref.Namespace != nil &&
+			ref.Group == ngfAPI.GroupName && ref.Kind == v1beta1.Kind("NginxProxy") {
+			nsName := types.NamespacedName{Name: ref.Name, Namespace: string(*ref.Namespace)}
+			return convertProxyConfig(nps[nsName], gwNsName)
+		}
+	}
+
+	return nil
+}
+
+func convertProxyConfig(ngfproxy *ngfAPI.NginxProxy, gwNsName string) *NginxProxy {
+	if ngfproxy != nil {
+		return validateProxyConfig(ngfproxy, gwNsName)
+	}
+	return nil
+}
+
+func validateProxyConfig(ngfproxy *ngfAPI.NginxProxy, gwNsName string) *NginxProxy {
+	var conds []conditions.Condition
+
+	var valErr []error
+
+	proxy := &NginxProxy{Source: ngfproxy}
+
+	if ngfproxy.Spec.HTTP != nil && ngfproxy.Spec.HTTP.Telemetry != nil &&
+		ngfproxy.Spec.HTTP.Telemetry.Tracing != nil {
+		ngfTracing := ngfproxy.Spec.HTTP.Telemetry.Tracing
+		tracing, valErr := validateTracing(ngfTracing, gwNsName)
+		if len(valErr) == 0 {
+			proxy.Tracing = tracing
+		}
+	}
+
+	for _, ve := range valErr {
+		conds = append(conds, staticConds.NewNginxProxyNotProgrammed(ve.Error()))
+	}
+
+	proxy.Conditions = conds
+	proxy.Valid = len(valErr) == 0
+
+	return proxy
+}
+
+const (
+	endpointRegex            = `^(?:(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,}|(?:\d{1,3}\.){3}\d{1,3}):\d{1,5}$` // nolint:lll
+	timeRegex                = `^(\d+y)??\s*(\d+M)??\s*(\d+w)??\s*(\d+d)??\s*(\d+h)??\s*(\d+m)??\s*(\d+s?)??\s*(\d+ms)??$`
+	endpointValidationString = "tracing.endpoint must be in the form `<IP>:<port>` or `<hostname>:<port>"
+)
+
+var (
+	reEndpoint = regexp.MustCompile(endpointRegex)
+	reTime     = regexp.MustCompile(timeRegex)
+)
+
+func validateTracing(ngfTracing *ngfAPI.Tracing, gwNsName string) (*Tracing, []error) {
```

Comment:
@pleshakov So is our plan to always do double validation for everything, due to the possibility that someone could alter the CRD? I get the thought behind it, though it's a shame that we aren't saving the effort on validation by relying on the CRD so we don't have to implement it ourselves.

Reply:
yep